core/jobs/batch_jobs/audit_topic_related_models_relation_jobs.py

```diff
@@ -104,23 +104,23 @@ def _validate_topic_models(
             for rights_model in topic_rights_models):
             errors.append(
                 base_validation_errors.ModelRelationshipError(
-                id_property=model_property.ModelProperty(
-                    topic_models.TopicModel,
-                    topic_models.TopicModel.name),
-                model_id=job_utils.get_model_id(topic_model),
-                target_kind='TopicRightsModel',
-                target_id=topic_model.name
+                    id_property=model_property.ModelProperty(
+                        topic_models.TopicModel,
+                        topic_models.TopicModel.name),
+                    model=topic_model,
```

Comment:
Generally it's better to pass around model ID than model ... why did this change?

In general, prefer to pass small things and primitive objects around the codebase, we try not to pass NDB models around in general.

Reply:
The issue is with the sub-error types and the way they are setup, see examples in this [file](https://github.com/oppia/oppia/blob/develop/core/jobs/types/base_validation_errors.py#L92). All the sub-types have the model being passed as they use different model properties. Like the one I linked needs to access to datetime fields. Since, the child types already receive model, is there a harm with passing it for the parent as well otherwise we need to modify all children to specifically pass id/kind. wdyt?

Reply:
Oh, interesting. I think we should have `__init__` of BaseAuditError and ModelRelationshipError use the same properties ideally, otherwise it's not really an "is a" relationship. (Currently they seem completely different so I don't know what's the point of the subclassing and it's kind of hard to reason about them.) Is it possible to do that?

Reply:
The issue here is that `ModelRelationshipError` needs to have extra properties about the target model for which the relationship error is generated. For example, if checking for a story summary model, we need information on which property in the story model refers to summary model. Further, it needs the ID and type of the target model or the target model itself to have that info in the error message:

```
message = (
    '%s=%s should correspond to the ID of an existing %s, but no such '
    'model exists' % (
        id_property, utils.quoted(target_id), target_kind))
```
Subclassing here is to keep the error message generation consistent. But the error message itself depends on another model, hence the `__init__` of `ModelRelationshipError` is different.

Reply:
I see, OK, thanks.

I think it's fine to change to `model`. It's not great though because we shouldn't be passing around models like this, so could you also please file an issue to, in the future, either use domain objects here (somehow, need to figure out how to get jobs to import from domain) or have a way to pass around models that are readonly? That way we at least have a todo to fix it (the main issue is with models getting passed around and possibly saved somewhere later on).

Reply:
Thanks, filed an issue: https://github.com/oppia/oppia/issues/22338 and added a TODO. I am happy to take this up as soon as I have more bandwidth. Expecting to have that post GSoC application period is complete and release tasks are done.

Reply:
OK, thanks Ankita. I think it's OK to do that in the longer term, the single-release-candidate stuff is more important and the design for this new issue is not that clear. So let's defer it to later (but thanks for filing the issue so that we don't forget!)